tests/lib/lte_lc_api/src/lte_lc_api_test.c

```diff
@@ -1982,7 +2079,95 @@ void test_lte_lc_neighbor_cell_measurement_gci(void)
 	test_event_data[0].type = LTE_LC_EVT_NEIGHBOR_CELL_MEAS;
 	test_event_data[0].cells_info.current_cell.mcc = 111;
 	test_event_data[0].cells_info.current_cell.mnc = 99;
-	test_event_data[0].cells_info.current_cell.id = 0x00112233;
+	test_event_data[0].cells_info.current_cell.id = LTE_LC_CELL_EUTRAN_ID_INVALID;
+	test_event_data[0].cells_info.current_cell.tac = 0x1A2B;
+	test_event_data[0].cells_info.current_cell.earfcn = 6200;
+	test_event_data[0].cells_info.current_cell.timing_advance = 64;
+	test_event_data[0].cells_info.current_cell.timing_advance_meas_time = 20877;
+	test_event_data[0].cells_info.current_cell.measurement_time = 189205;
+	test_event_data[0].cells_info.current_cell.phys_cell_id = 110;
+	test_event_data[0].cells_info.current_cell.rsrp = 53;
+	test_event_data[0].cells_info.current_cell.rsrq = 22;
+	test_event_data[0].cells_info.ncells_count = 0;
+	test_event_data[0].cells_info.gci_cells_count = 2;
+
+	test_gci_cells[0].mcc = 111;
+	test_gci_cells[0].mnc = 98;
+	test_gci_cells[0].id = 0x00567812;
+	test_gci_cells[0].tac = 0x3C4D;
+	test_gci_cells[0].earfcn = 1300;
+	test_gci_cells[0].timing_advance = 65535;
+	test_gci_cells[0].timing_advance_meas_time = 4;
+	test_gci_cells[0].measurement_time = 189241;
+	test_gci_cells[0].phys_cell_id = 75;
+	test_gci_cells[0].rsrp = 53;
+	test_gci_cells[0].rsrq = 16;
+
+	test_gci_cells[1].mcc = 112;
+	test_gci_cells[1].mnc = 97;
+	test_gci_cells[1].id = 0x0011AABB;
+	test_gci_cells[1].tac = 0x5E6F;
+	test_gci_cells[1].earfcn = 2300;
+	test_gci_cells[1].timing_advance = 65534;
+	test_gci_cells[1].timing_advance_meas_time = 5;
+	test_gci_cells[1].measurement_time = 189245;
+	test_gci_cells[1].phys_cell_id = 449;
+	test_gci_cells[1].rsrp = 51;
+	test_gci_cells[1].rsrq = 11;
+
+	at_monitor_dispatch(at_notif);
+}
+
+void test_lte_lc_neighbor_cell_measurement_gci_status_failed(void)
+{
+	int ret;
+	struct lte_lc_ncellmeas_params params = {
+		.search_type = LTE_LC_NEIGHBOR_SEARCH_TYPE_GCI_EXTENDED_LIGHT,
+		.gci_count = 10,
+	};
+	strcpy(at_notif,
+		"%NCELLMEAS: 1,"
+		"\"1FFFFFFF\",\"11199\",\"1A2B\",64,20877,6200,110,53,22,189205,1,0,"
+		"\"00567812\",\"11198\",\"3C4D\",65535,4,1300,75,53,16,189241,0,0,"
+		"\"0011AABB\",\"11297\",\"5E6F\",65534,5,2300,449,51,11,189245,0,0\r\n");
+
+	lte_lc_callback_count_expected = 1;
+
+	__mock_nrf_modem_at_printf_ExpectAndReturn("AT%NCELLMEAS=4,10", EXIT_SUCCESS);
+
+	ret = lte_lc_neighbor_cell_measurement(&params);
+	TEST_ASSERT_EQUAL(EXIT_SUCCESS, ret);
+
+	test_event_data[0].type = LTE_LC_EVT_NEIGHBOR_CELL_MEAS;
+	test_event_data[0].cells_info.current_cell.id = LTE_LC_CELL_EUTRAN_ID_INVALID;
+
+	at_monitor_dispatch(at_notif);
+}
+
+void test_lte_lc_neighbor_cell_measurement_gci_status_incomplete(void)
+{
+	int ret;
+	struct lte_lc_ncellmeas_params params = {
+		.search_type = LTE_LC_NEIGHBOR_SEARCH_TYPE_GCI_EXTENDED_LIGHT,
+		.gci_count = 10,
+	};
+	strcpy(at_notif,
+		"%NCELLMEAS: 2,"
```

Comment:
This is not related to the tests, but it's not very nice that it's not visible in the LTE LC API that the measurement was interrupted. There's no clean way to add a status to the current API, I'm afraid.

Reply:
That's the way it's been done.